Feature/router

TemVin opened the pull request.

TemVin committed:
add change

-- add new field 'imgUrl'

TemVin committed:
add change

--  add route to page book

TemVin committed:
add change

--Added new page 'BookCard' to display a particular book

TemVin committed:
add change

-- 'imgUrl' variable added to other components and pages, renamed component 'Book' to 'Card'

TemVin committed:
add change

-- removed title 'React Starter Boilerplate'

TemVin committed:
add change

-- added URL change while searching (when pressing enter)

TemVin committed:
add change

--search text is displayed in url

TemVin committed:
add change

--added validation

TemVin committed:
add change

--added link to homepage

TemVin committed:
add change

-- add validation

TemVin committed:
add changes

-- added new classes

TemVin committed:
add changes

-- css styling added

TemVin committed:
add change

-- removed extra space

TemVin committed:
add change

-- added new color

TemVin committed:
add change

-- changed tab name

TemVin committed:
add change

-- removed  tag 'hr'

TemVin committed:
add change

-- install 'classnames'

konovalho commented on src/actions/books.js:
```diff
@@ -1,13 +1,13 @@
 
-
-export const addBook = (id, title, author, description) => ({
+export const addBook = (id, title, author, description, imgUrl) => ({
```
Опять.
Для чего кидать каждое поле отдельно?
Кидай сразу весь объект

TemVin replied:
Есть вопрос.

konovalho replied:
задавай)

TemVin replied:
задал
в итоге const book = {
        id,
        title: fields[0].value,
        author: fields[1].value,
        description: fields[2].value,
        imgUrl: fields[3].value,
      };
addBook(book);

konovalho commented on src/components/Card/Card.jsx:
```diff
@@ -83,33 +84,41 @@ class Book extends PureComponent {
-            <br />
-            <button styleName="delete" onClick={this.handleClick}>Удалить</button>
+        <div styleName="container">
+          <div styleName="bookCard">
```
просто `card` приставку book ни к чему испольщовать так как у нас react-css-modules

TemVin replied:
исправил

konovalho commented on src/components/Card/Card.scss:
```diff
@@ -1,43 +1,55 @@
+
+.container {
+  float: left;
+  width: 33.33%;
```
лучше `calc(100% / 3)`

TemVin replied:
добавил)

konovalho commented on src/components/Card/Input/Input.jsx:
```diff
@@ -9,6 +9,7 @@ class Input extends PureComponent {
     return (
       <div styleName="root">
```
получается что div лишний, можно сразу написать input

TemVin replied:
убрал)

konovalho commented on src/components/SearchInput/SearchInput.jsx:
```diff
@@ -9,9 +9,17 @@ class SearchInput extends PureComponent {
+        <div styleName="container">
+          <img
+            styleName="icon"
+            src="https://api.icons8.com/download/0c1134f592bf842c2b6b34fc071b57f23d397885/windows10/PNG/512/Very_Basic/search-512.png"
```
скачай картинку и добавь в папку images

TemVin replied:
Добавил

konovalho commented on src/pages/BookCard/BookCard.jsx:
```diff
@@ -0,0 +1,46 @@
+    return book.map(item => (
+      <div styleName="root">
+        <div key={item.id} styleName="card">
+          <div styleName="cardOptions">
```
просто `options`

TemVin replied:
исправил

konovalho commented on src/pages/BookCard/BookCard.jsx:
```diff
@@ -0,0 +1,46 @@
+            <span>Автор: {item.author}</span> <br />
+            <span>Описание: {item.description}</span> <br />
+          </div>
+          <div styleName="blockImage">
```
можно без `block` и дальше

TemVin replied:
хорошо)

konovalho commented on src/pages/HomePage/BooksContainer/BooksContainer.jsx:
```diff
@@ -4,20 +4,20 @@ import { bindActionCreators } from 'redux';
     books.map(item => (
-      <Book
+      <Card
         removeBook={this.props.removeBook}
```
насчет порядка передачи пропсов.
Нужный порядок: 
1) key
2) id
3) пропсы с примитивами
4) функции

TemVin replied:
запомнил)

konovalho commented on src/pages/HomePage/SearchContainer/SearchContainer.jsx:
```diff
@@ -1,30 +1,45 @@
+  updateUrl(valueInput) {
+    const { location, history } = this.props;
+    if (valueInput) {
+      location.search = `/?search=${valueInput}`;
```
ты меняешь объект по ссылке, это грубейшая ошибка, говорил тебе об этом.
Создай новую переменную.
Зачем тебе вообще location здесь? если ты не используешь его данные

TemVin replied:
создал переменную searchUrl
 let searchUrl;
    if (valueInput) {
      searchUrl = `/?search=${valueInput}`;
    }
    history.push(searchUrl);

TemVin committed:
fixed code review

-- removed unnecessary tags, corrected class names, added a picture to a folder 'images', corrected the order of the transfer of props, corrected change url when searching, moved the container 'SearchContainer' below 'FormContainer'

konovalho commented on src/components/Card/Card.jsx:
```diff
@@ -85,7 +85,7 @@ class Card extends PureComponent {
         <div styleName="container">
-          <div styleName="bookCard">
+          <div styleName="book">
             <div styleName="bookOptions">
```
а что `bookOptions ` чем то отличается?

TemVin replied:
изменил на card. Я правильно понял твой вопрос?)

konovalho commented on src/pages/HomePage/SearchContainer/SearchContainer.jsx:
```diff
@@ -12,13 +12,12 @@ import styles from './SearchContainer.scss';
+      searchUrl = `/?search=${valueInput}`;
     }
-    history.push(location.search);
+    history.push(searchUrl);
```
то есть если в if не зайдет ты пушниш в history undefined?

TemVin replied:
Да, ты прав)
хоть undefined не отображается, но неправильно)

TemVin committed:
changed book addition

-- created an object and pass it

TemVin committed:
change search

-- don't push into history 'undefined'

TemVin committed:
changed styles css
